levelCreator/states/create.js

```diff
@@ -97,6 +116,31 @@ function initCreateState() {
         tile: game.activeTool
       };
     }
+
+    function move(xDiff, yDiff) {
+      const clickPoint = new Phaser.Point(game.camera.width / 2 - xDiff, game.camera.height / 2 - yDiff);
+      game.dolly.targetPos = game.dolly.screenspaceToWorldspace( clickPoint );
+    }
+
+    function rotate(dir) {
+      if(Date.now() - lastRotTime > 500) {
+        if (!game.dolly.targetAng) game.dolly.targetAng = 0;
+        game.dolly.targetAng += dir * Math.PI / 2;
+        lastRotTime = Date.now();
+      }
+    }
+
+    const moveAmount = 64;
+
+    if (upKey.isDown) move(0, moveAmount);
+    if (downKey.isDown) move(0, -moveAmount);
+    if (leftKey.isDown) move(moveAmount, 0);
+    if (rightKey.isDown) move(-moveAmount, 0);
```

Comment:
If we do what I mentioned earlier, we could add a method to the object that returns the vector of motion based on keys pressed ( xmotion = Number( this.up.isDown ) - Number( this.down.isDown ) ) and not have to call the move function 4 times.